### PR TITLE
nodejs pub/sub

### PR DESCRIPTION
**Pull request by ajibade3210**
Please read through the [writing guide](https://reflectoring.io/contribute/writing-guide/) before submitting a pull request and go through this checklist (at least for your first article).

**Feel free to raise the pull request even if the checklist hasn't been worked through, yet**, to take advantage of the online preview that is generated for each pull request (see "deploy/netlify" under "Show all checks" in your pull request). But please only flag it to be reviewed by your editor after going through the checklist.

## Review
- [ ] you have reviewed the text a day after finishing it (you'd be surprised how many errors you can avoid by just getting some distance to what you've written)

## Content
- [ ] the article explains *why* and not only *how*
- [ ] (half-)sentences that express a meaningful idea that is understandable by itself own are **bold** to allow quick scanning of the text
- [ ] the text has been checked with [Grammarly](https://grammarly.com) (the free tier is enough)

## Text
- [ ] the text is conversational ("we" instead of "you", easy language)
- [ ] there are no "walls of text" (short sentences, short paragraphs with a single idea)
- [ ] the text is inclusive ("the developers" and "they" instead of "the developer" and "he/she")
- [ ] the text uses active voice instead of passive voice
- [ ] names are spelled consistently throughout the article (correct uppercase / lowercase)
- [ ] file names and variables are highlighted as `code`

## Structure
- [ ] the article has a short intro before the first headline
- [ ] by reading the table of contents (TOC) alone, I get a sense of the "story" of the article (check the TOC in the preview after raising the PR or by starting the blog locally on your machine)
- [ ] if using sub-headlines, there's at least two sub-headlines below a main-headline
- [ ] headlines are consistent throughout the article (same level of abstraction, same style - if one headline starts with a verb in imperative form, the other headlines on the same level should probably, too)

## Code Examples
- [ ] code examples are introduced with a ":" and explained below the code example
- [ ] code examples are formatted to avoid horizontal scrolling (manual line breaks and 2-spaces indentation)

### CODE REVIEW

**Comment by ajibade3210** [comment]
@thombergs season's greetings. 
I figure you have been busy, so I'm just dropping this here as a reminder for this pR to help review whenever you get a chance.

Many thanks

**Review by thombergs — CHANGES_REQUESTED** [review]

File: content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+To tackle these issues, we can switch to asynchronous communication using a Pub/Sub pattern. Pub/Sub allows systems to work independently, solving issues like unavailability and queuing. It also introduces flexibility and scalability to our systems.
+
+Using Pub/Sub, Sever A (Publisher) publishes events related to changes, categorizing them as a topic, and Sever B (Subscriber) subscribes to these events. The publisher-subscription model, allows messages to be disseminated to all subscriptions associated with a specific topic.
+
+This approach provides flexibility to subscribers like Sever B, who can choose to pull messages at their convenience or have messages pushed to a specified endpoint. The push mechanism ensures proactive message delivery, creating a dynamic and adaptable communication framework for diverse application needs.
+
+For a deeper understanding of Pub/Sub, let's explore some of its key terminology and common use cases.
+
+### Google Cloud Pub/Sub Terminology
+![Google Pub/Sub flow](https://cloud.google.com/static/pubsub/images/pub_sub_flow.svg)
```
[the remark this answers is not in the document]
Comment (ajibade3210) [reply content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md]: Okay thank you i have changed the Visual

File: content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+  Input the topic name `user_creation` and click the **CREATE** button.
+  {{% image alt="create-gcps-topic-2" src="images/posts/nodejs-pub-sub/create-gcps-topic-2.png" %}}
+
+  This action will create our `user_creation` topic and a default subscription named `user_creation-sub` will be automaticallygenerated since we left the **Add default subscription** box checked.
+   {{% image alt="default-subcription" src="images/posts/nodejs-pub-sub/default-subcription.png" %}}
+
+- Subscriptions define how messages are delivered to subscribers, and specifying the delivery type is crucial for receiving real-time updates. Next, we will create two subscriptions with delivery types `pull` and `push` for our subscribers to connect with a topic. 
+
+To create this custom subscription, click on **CREATE SUBSCRIPTION** button then to:
```
[the remark this answers is not in the document]
Comment (ajibade3210) [reply content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md]: Thank You

File: content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+
+#### Receive Message Via Pull
+To verify message publication, our Email service can make an API call to the `/api/email/pull` route using the pull subscription.
+{{% image alt="receive-pull-message" src="images/posts/nodejs-pub-sub/receive-pull-message.png" %}}
+
+After making this call, check the Email service terminal; we should be able to see the received data in our terminal logs.
+{{% image alt="receive-message-terminal" src="images/posts/nodejs-pub-sub/receive-message-terminal.png" %}}
+
+#### Receiving Message Via Push
+To receive push messages, we will utilize an HTTPS endpoint for our webhook. You can achieve this with either a live endpoint or, if using local routes, employ Ngrok to expose them.
```
[the remark this answers is not in the document]
Comment (ajibade3210) [reply content/blog/2023/2023-12-11-google-pub-sub-in-node.js.md]: Okay Noted :)

**Comment by ajibade3210** [comment]
@thombergs I have updated this please help review

**Pull request merged**